### ISSUE TITLE
Uploading to a newly created folder

### ISSUE THREAD

**Issue by JeffMorison** [vgrem/Office365-REST-Python-Client#256]
I'm trying to upload to a newly created folder (in the Temp directory) and the only way I can achieve this is with the following but I don't understand why I can't use the `archive_folder` directly without calling `get_folder_by_server_relative_url` a second time:

```
temp_folder = ctx.web.get_folder_by_server_relative_url("Temp")
archive_folder = temp_folder.folders.add("Archive_")
ctx.execute_query()

target_folder = ctx.web.get_folder_by_server_relative_url("Temp/Archive_")
target_file = target_folder.upload_file(name, file_content)
ctx.execute_query()
```

This gives an error:

```
temp_folder = ctx.web.get_folder_by_server_relative_url("Temp")
archive_folder = temp_folder.folders.add("Archive_")
ctx.execute_query()

target_file = archive_folder.upload_file(name, file_content)
ctx.execute_query()
```

Error:
```
('-2147024891, System.UnauthorizedAccessException', 'Access denied. You do not have permission to perform this action or access this resource.', "403 Client Error: Forbidden for url:
```

**Comment by beliaev-maksim** [comment]
@JeffMorison 
I added new example how to upload a file [create_folders](https://github.com/vgrem/Office365-REST-Python-Client/blob/master/examples/sharepoint/folders/create_folders.py)

you can use `ensure_folder_path` and then just use function `.upload_file`


@vgrem 
I think this one could be closed

**Comment by vgrem** [comment]
Greetings "

**Comment by vgrem** [comment]
Greetings @JeffMorison,

thank you for catching this issue and reporting it! I confirm it indeed was caused by `Folder.upload_file` method, in particular in some cases, _file server relative url_ was incorrectly determined from _folder name_

The fix will be included in the next release (`2.2.3`),  meanwhile  it could be grabbed via GitHub:

`pip install git+https://github.com/vgrem/Office365-REST-Python-Client.git`

**Comment by spurthikaribasaiah** [comment]
Hi @vgrem ,

I am getting error "module named adal not found" when trying to import libraries from 0ffice365. This was waorking fine earlier. allof sudden this has stopped.
Please help me in resolving this
